### PR TITLE
#107 test cases 

### PR DESCRIPTION
**Pull request by maddygriz**
The test cases for GeomxTools are finished except #115 and the DA comparison

### CODE REVIEW

File: tests/testthat/test_normalization.R [hunk 1]
```diff
@@ -141,16 +167,33 @@ test_that("hk norm values are correct", {
...
 # #### req 6 verify calculation of subtract background norm factors
 # #call subtract bg normalization
-
 test_that("background subtraction throws warning on probe level data", {
   expect_warning(normalize(demoData , data_type="RNA",
                            norm_method="subtractBackground", fromElt="exprs", toElt="bg_norm"))
 })
 
 
+target_demoData <- normalize(target_demoData , data_type="RNA",
```
Comment (NicoleEO) [inline tests/testthat/test_normalization.R:180]: Repeat of L197-211

File: tests/testthat/test_utils.R [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  expect_identical(logtBase(fakeCountsWith0_NA), c(log(thresholdValues(fakeCountsWith0_NA), base = 2), NA))
+  expect_identical(logtBase(fakeCountwith0), log(thresholdValues(fakeCountwith0), base = 2))
+})
+
+
+#These functions are tested in test_shiftCountsOne.R:
+#shiftCountsOne()
+
+#These functions are tested in test_GeoMxSet_probeCollapse.R:
+#collapseCounts()
```
Comment (NicoleEO) [inline tests/testthat/test_utils.R:66]: Add line at end

File: tests/testthat/test_utils.R [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  expect_identical(logtBase(fakeCountsWithNA), log(thresholdValues(fakeCountsWithNA), base = 2))
+  expect_identical(logtBase(fakeCountsWith0_NA), c(log(thresholdValues(fakeCountsWith0_NA), base = 2), NA))
+  expect_identical(logtBase(fakeCountwith0), log(thresholdValues(fakeCountwith0), base = 2))
+})
+
+
+#These functions are tested in test_shiftCountsOne.R:
+#shiftCountsOne()
+
+#These functions are tested in test_GeoMxSet_probeCollapse.R:
```
Comment (NicoleEO) [inline tests/testthat/test_utils.R:65]: Delete this line as this function is not used at all. We should mark it out as not tested or used

File: tests/testthat/test_writeNanoStringGeoMxSet.R [hunk 4]
```diff
@@ -41,13 +41,37 @@ testthat::test_that("test DCC files are written after read in", {
...
+                                          phenoDataSheet = "CW005",
+                                          phenoDataDccColName = "Sample_ID",
+                                          protocolDataColNames = c("aoi",
+                                                                   "cell_line",
+                                                                   "roi_rep",
+                                                                   "pool_rep",
+                                                                   "slide_rep"),
+                                          experimentDataColNames = c("panel")))
+
+# req 2: test GeomxSet object from written DCC is identical to original
```
Comment (NicoleEO) [inline tests/testthat/test_writeNanoStringGeoMxSet.R:57]: I like this test. Nice addition.

**Review by NicoleEO — CHANGES_REQUESTED** [review]

File: tests/testthat/test_utils.R (hunk 3 above)
Comment (maddygriz) [reply tests/testthat/test_utils.R:65]: I marked the function in utils.R as NOT TESTED OR USED

**Review by NicoleEO — APPROVED** [review]

**Pull request merged**